File: ChickInRun/Assets/SubModel/san_lava/LavaGate.js
#pragma strict

private var lGate : GameObject;
public var lavaParticle : GameObject;


function Start(){
	lGate = GameObject.FindGameObjectWithTag("LavaGate");
}

function OnTriggerEnter (col : Collider){

	if(col.gameObject.tag == "Chick"){
		lGate.GetComponent(GateOpen).openNum++;
		Instantiate(lavaParticle, transform.position, transform.rotation);
		Destroy(gameObject);
		lGate.GetComponent(GateOpen).lsoundCheck = true;
	}
}

function Update(){

	transform.rotation *=Quaternion.AngleAxis(2, Vector3(0,0,1));

}

File: ChickInRun/Assets/Script/LavaGate.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LavaGate : MonoBehaviour
{
    private GameObject lGate;
    public GameObject lavaParticle;

    // Start is called before the first frame update
    void Start()
    {
        lGate = GameObject.FindGameObjectWithTag("LavaGate");
    }

    // Update is called once per frame
    void Update()
    {
        transform.rotation *= Quaternion.AngleAxis(2.0f, new Vector3(0.0f, 0.0f, 1.0f));
    }

    private void OnTriggerEnter(Collider other)
    {
        if(other.gameObject.tag == "Chick")
        {
            lGate.GetComponent<GateOpen>().openNum++;
            Instantiate(lavaParticle, transform.position, transform.rotation);
            Destroy(gameObject);
            lGate.GetComponent<GateOpen>().lsoundCheck = true;
        }
    }
}
